File: repr/repr.cpp
#include <iostream>

template <typename T, typename U> struct Foo {
  short count;
  T data1;
  U data2;

  Foo(short count, T data1, U data2)
      : count(count), data1(data1), data2(data2) {}
};

int main() {

    auto foo1 = Foo<short, int>(1, 1, 1);
    auto foo2 = Foo<int, short>(1, 1, 1);

    std::cout << sizeof(foo1) << ' ' << sizeof(foo2) << '\n';

    return 0;
}

File: repr/repr.rs
use std::mem;

struct Foo<T, U> {
    count: u16,
    data1: T,
    data2: U,
}

fn main() {
    let foo1: Foo<u16, u32> = Foo {
        count: 1,
        data1: 1,
        data2: 1,
    };
    let foo2: Foo<u32, u16> = Foo {
        count: 1,
        data1: 1,
        data2: 1,
    };

    println!("{} {}", mem::size_of_val(&foo1), mem::size_of_val(&foo2));
}
